Packs/FeedGithub/Integrations/FeedGithub/FeedGitHub.py

```diff
@@ -0,0 +1,514 @@
+        relevant_files = parsing_files_by_status(all_commits_files)
+        return relevant_files, current_repo_head # type: ignore
+
+    except IndexError:
+        return [], last_commit_fetch
+
+
+def parse_and_map_yara_content(content_item: str) -> list:
+    """
+    Parses YARA rules from a given content item and maps their attributes.
+
+    Args:
+        content_item (str): A string containing one or more YARA rules.
+
+    Returns:
+        list: A list of dictionaries representing parsed and mapped YARA rules.
+              Each dictionary contains attributes such as rule name, description, author, etc.
+    """
+    pattern = re.compile(r"rule\s+\w+\s*?\{(?:.*?\n)*?\}", re.DOTALL)
+    content_rules = pattern.findall(content_item)
+    parsed_rules = []
+    for rule in content_rules:
+        parser = plyara.Plyara()
+        try:
+            parsed_rule = parser.parse_string(rule)[0]
+            metadata = {key: value for d in parsed_rule["metadata"] for key, value in d.items()}
+            mapper = {
+                "value": parsed_rule["rule_name"],
+                "description": metadata.get("description", ""),
+                "author": metadata.get("author", ""),
+                "rulereference": metadata.get("reference", ""),
+                "sourcetimestamp": metadata.get("date", ""),
+                "id": metadata.get("id", ""),
+                "rule strings": parsed_rule.get("strings", []),
+                "condition": " ".join(parsed_rule["condition_terms"]),
+                "type": "YARA",
+                "raw rule": rule,
+            }
+            
+            parsed_rules.append(mapper)
+        except Exception as e:
+            demisto.log(f"Rull: {rule} cannot be processed. Error Message: {e}")
+            continue
+    return parsed_rules
+
+
+def get_yara_indicator(content: list):
+    """
+    Retrieves YARA indicators from a list of content items.
+
+    Args:
+        content (list): A list of strings containing YARA rules.
+
+    Returns:
+        list: A list of dictionaries representing parsed and mapped YARA rules for each content item.
+    """
+    res = []
+    for item in content:
+        res += parse_and_map_yara_content(item)
+    
+    return res
+
+
+def detect_domain_type(domain: str):
+    """
+    Detects the type of an indicator (e.g., Domain, DomainGlob) using tldextract library.
+
+    Args:
+        domain (str): The indicator value to be analyzed.
+
+    Returns:
+        Optional[FeedIndicatorType]: The type of the indicator, or None if detection fails.
+    """
+    try:
+        import tldextract
+    except Exception:
+        raise Exception(
+            "Missing tldextract module, In order to use the auto detect function please use a docker"
+            " image with it installed such as: demisto/jmespath"
+        )
+    try:
+        tldextract_version = tldextract.__version__
+        if LooseVersion(tldextract_version) < "3.0.0":
```

Comment:
we dont need to do any of these version checks. CSP is the only place that is relevant